pkg/controller/elasticsearch/client/v8.go

```diff
@@ -35,6 +35,28 @@ func (c *clientV8) GetClusterState(ctx context.Context) (ClusterState, error) {
 	return response, err
 }
 
+func (c *clientV8) CreateCrossClusterAPIKey(ctx context.Context, request CrossClusterAPIKeyCreateRequest) (CrossClusterAPIKeyCreateResponse, error) {
+	var response CrossClusterAPIKeyCreateResponse
+	err := c.post(ctx, "/_security/cross_cluster/api_key", request, &response)
+	return response, err
+}
+
+func (c *clientV8) UpdateCrossClusterAPIKey(ctx context.Context, apiKeyID string, request CrossClusterAPIKeyUpdateRequest) (CrossClusterAPIKeyUpdateResponse, error) {
+	var response CrossClusterAPIKeyUpdateResponse
+	err := c.put(ctx, fmt.Sprintf("/_security/cross_cluster/api_key/%s", apiKeyID), request, &response)
+	return response, err
+}
+
+func (c *clientV8) GetCrossClusterAPIKeys(ctx context.Context, name string) (CrossClusterAPIKeyList, error) {
+	var response CrossClusterAPIKeyList
+	err := c.get(ctx, fmt.Sprintf("/_security/api_key?active_only=true&name=%s", name), &response)
+	return response, err
+}
+
+func (c *clientV8) InvalidateCrossClusterAPIKey(ctx context.Context, name string) error {
```

Comment:
Because the key is actually not deleted:

> This API invalidates API keys created by the [create API key](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-create-api-key.html) or [grant API key](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-grant-api-key.html) APIs. Invalidated API keys fail authentication, but they can still be viewed using the [get API key information](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-get-api-key.html) and [query API key information](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-query-api-key.html) APIs, for at least the [configured retention period](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html#api-key-service-settings-delete-retention-period), until they are automatically deleted.

https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-invalidate-api-key.html

Reply:
TIL